server/src/main/java/com/objectcomputing/checkins/services/memberprofile/currentuser/CurrentUserServicesImpl.java

```diff
@@ -0,0 +1,39 @@
+package com.objectcomputing.checkins.services.memberprofile.currentuser;
+
+import com.objectcomputing.checkins.services.memberprofile.MemberProfile;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileRepository;
+import com.objectcomputing.checkins.services.memberprofile.MemberProfileServicesImpl;
+
+import javax.inject.Inject;
+import javax.validation.constraints.NotNull;
+import java.time.LocalDate;
+import java.util.Map;
+import java.util.Optional;
+
+public class CurrentUserServicesImpl implements CurrentUserServices {
+
+    @Inject
+    MemberProfileRepository memberProfileRepo;
+
+    @Inject
+    MemberProfileServicesImpl memberProfileServicesImpl;
+
+    @Override
+    public MemberProfile findOrSaveUser(@NotNull Map<String, Object> tokenAttributes) {
+
+        String workEmail = tokenAttributes.get("email").toString();
+
+        if (workEmail == null) {
+            throw new InvalidUserException(String.format("Invalid work email"));
+        } else {
+
+            Optional<MemberProfile> userProfile = memberProfileRepo.findByWorkEmail(workEmail);
```

Comment:
this will be memberProfileServiceImpl.findByWorkEmail once the open PR is merged